lib/ansible/modules/cloud/digital_ocean/digital_ocean_droplet.py

```diff
@@ -0,0 +1,340 @@
+            self.module.fail_json(changed=False, msg=json_data['message'])
+        if self.wait:
+            json_data = self.ensure_power_on(json_data['droplet']['id'])
+            droplet_data = self.get_addresses(json_data)
+        self.module.exit_json(changed=True, data=droplet_data)
+
+    def delete(self):
+        json_data = self.get_droplet()
+        if json_data:
+            if self.module.check_mode:
+                self.module.exit_json(changed=True)
+            response = self.rest.delete('droplets/{0}'.format(json_data['droplet']['id']))
+            json_data = response.json
+            if response.status_code == 204:
+                self.module.exit_json(changed=True, msg='Droplet deleted')
+            self.module.fail_json(changed=False, msg='Failed to delete droplet')
+        else:
+            self.module.exit_json(changed=False, msg='Droplet not found')
+
+    def ensure_power_on(self, droplet_id):
+        end_time = time.time() + self.wait_timeout
+        while time.time() < end_time:
+            response = self.rest.get('droplets/{0}'.format(droplet_id))
+            json_data = response.json
+            if json_data['droplet']['status'] == 'active':
+                return json_data
+            time.sleep(min(2, end_time - time.time()))
+        self.module.fail_json(msg='Wait for droplet powering on timeout')
+
+
+def core(module):
+    state = module.params.pop('state')
+    droplet = DODroplet(module)
+    if state == 'present':
+        droplet.create()
+    elif state == 'absent':
+        droplet.delete()
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(choices=['present', 'absent'], default='present'),
+            oauth_token=dict(
+                aliases=['API_TOKEN'],
+                no_log=True,
+                fallback=(env_fallback, ['DO_API_TOKEN', 'DO_API_KEY', 'DO_OAUTH_TOKEN'])
+            ),
+            name=dict(type='str'),
+            size=dict(aliases=['size_id']),
+            image=dict(aliases=['image_id']),
+            region=dict(aliases=['region_id']),
+            ssh_keys=dict(type='list'),
+            private_networking=dict(type='bool', default=False),
+            backups=dict(type='bool', default=False),
+            monitoring=dict(type='bool', default=False),
+            id=dict(aliases=['droplet_id'], type='int'),
+            user_data=dict(default=None),
+            ipv6=dict(type='bool', default=False),
+            volumes=dict(type='list'),
+            tags=dict(type='list'),
+            wait=dict(type='bool', default=True),
+            wait_timeout=dict(default=120, type='int'),
+            unique_name=dict(type='bool', default=False),
+        ),
+        required_one_of=(
+            ['id', 'name'],
+        ),
+        required_if=([
+            ('state', 'present', ['name', 'size', 'image', 'region']),
+        ]),
+        supports_check_mode=True,
+    )
+
+    try:
```

Comment:
catchall exceptions make for bad UI, we normally prefer a user friendly message, detailing what failed and hinting possible resolutions vs passing on API messages that are normally only useful to programmers

Reply:
I added https://github.com/ansible/ansible/blob/83df7088e1d13e84caec7f06495b1324ea1a7cd7/lib/ansible/modules/cloud/digital_ocean/digital_ocean_droplet.py#L258

to handle any errors on create. This catches invalid image names, regions, and sizes.

I realize this probably isn't what you wanted based on your comment about but in all of these cases, DigitalOcean returns an HTTP 422 + a message. Short of parsing the message itself or displaying a generic error message, I just relay it back to the user. IMO, the messages that come from DigitalOcean describe what failed in a user-friendly way.